packages/mgt-components/src/components/mgt-picker/mgt-picker.ts

```diff
@@ -247,14 +247,28 @@ export class MgtPicker extends MgtTemplatedComponent {
         placeholder=${this.placeholder}>
           ${this.response.map(
             item => html`
-            <fluent-option value=${item.id} @click=${(e: MouseEvent) => this.handleClick(e, item)}> ${
-              item[this.keyName]
-            } </fluent-option>`
+            <fluent-option value=${item.id} @click=${(e: MouseEvent) =>
+              this.handleClick(e, item)}> ${this.getNestedPropertyValue(item, this.keyName)} </fluent-option>`
           )}
       </fluent-combobox>
      `;
   }
 
+  private getNestedPropertyValue(item: Entity, keyName: string) {
+    const keys = keyName.split('.');
+    let value: Entity | object | string = item;
+
+    for (const key of keys) {
+      value = value[key] as object | string;
+
+      if (value === undefined) {
+        return '';
```

Comment:
Let's put a console warning here to help developers understand why there is no visible string in the option

Reply:
I have now added a console warning